### PR TITLE
bpo-31128: Allow pydoc to bind to arbitrary hostnames

### PR DESCRIPTION
**Pull request by feanil**
https://bugs.python.org/issue31128

### CODE REVIEW

**Comment by the-knights-who-say-ni** [comment]
Hello, and thanks for your contribution!

I'm a bot set up to make sure that the project can legally accept your contribution by verifying you have signed the [PSF contributor agreement](https://www.python.org/psf/contrib/contrib-form/) (CLA).

Unfortunately we couldn't find an account corresponding to your GitHub username on [bugs.python.org](http://bugs.python.org/) (b.p.o) to verify you have signed the CLA (this might be simply due to a missing "GitHub Name" entry in your b.p.o account settings). This is necessary for legal reasons before we can look at your contribution. Please follow [the steps outlined in the CPython devguide](https://devguide.python.org/pullrequest/#licensing) to rectify this issue.


Thanks again to your contribution and we look forward to looking at it!

**Review by rajathagasthya — APPROVED** [review]
LGTM. 

Verified locally:
```
~/dev/cpython $ ./python.exe -m pydoc -h localhost -p 5000
Server ready at http://localhost:5000/
Server commands: [b]rowser, [q]uit
server>
```

File: Lib/pydoc.py
Comment (merwok) [inline Lib/pydoc.py:2254]: If this class is considered part of the module’s public API, this change would break compatibility.  Alternative ways to achieve the same result could be:

* add host after the existing params, with default value to keep compat
* allow the existing port to be a (host, port) tuple

Comment (feanil) [reply Lib/pydoc.py:2254]: @merwok,  I considered that but figured it was a class inside a underscored method so its not guaranteed to remain compatible.  I figured it would be better to opt for the more readable signature instead of being backwards compatible.  If it's a blocker to make this compatible let me know but if we're implying that this function should be treated as if it is public we may want to remove the prefixed underscore but that will definitely break people.

Comment (merwok) [reply Lib/pydoc.py:2254]: Ah!  Thanks for noting these classes are inside a private method, that excludes compat considerations.  Patch looks fine.

**Comment by Mariatta** [comment]
@merwok Aren't you a Core Developer? So this should not be "awaiting core review" anymore. :thinking:

**Comment by Mariatta** [comment]
Oh I see, I think it's because you did not "approve" the PR yet ...

**Comment by merwok** [comment]
I used to be, but I’m not part of the github team :)

File: Lib/test/test_pydoc.py
Comment (merwok) [inline Lib/test/test_pydoc.py]: Technically this test does not validate that the patch is correct, since it would also pass without the code change!  But it’s probably fine for this patch.

**Review by merwok — APPROVED** [review]

**Comment by Mariatta** [comment]
@merwok I see that you are still a committer in b.p.o. I hope you'll continue contributing as core developer 😄 If interested, perhaps check with @brettcannon of what needs to happen?

**Comment by brettcannon** [comment]
I just need @merwok to tell me he wants to be a core dev again, then I can add him on GitHub. 😄

**Comment by feanil** [comment]
Checking back in on this, I just wanted to confirm that there is nothing left to do on my end for this, and that it's just waiting for someone on the core team to review and merge(hopefully)?

**Comment by brettcannon** [comment]
@feanil yep, the `awaiting core review` label means you're waiting on a core dev to review this and either ask for changes or merge it.

File: Lib/pydoc.py
Comment (brettcannon) [inline Lib/pydoc.py]: Using `-h` is a bit dodgy since it's so commonly used as shorthand for `--help`. Unfortunately I don't have a better suggestion beyond `-n`.

File: Lib/test/test_pydoc.py
Comment (brettcannon) [reply Lib/test/test_pydoc.py]: Is `localhost` specified anywhere else in the `pydoc.py` file? If it isn't then while you're right it isn't testing a difference from the default from the public API, this does help make sure that at least the value is being used.

**Review by brettcannon — CHANGES_REQUESTED** [review]
The code looks good, just a question of whether the CLI switch is the right letter.

Also don't forget to add yourself to `Misc/ACKS` and to write a news entry. 😄

**Comment by merwok** [comment]
@brettcannon Alright please add me to the team!  I can at least review distutils PRs since not many other people would do that, and hopefully get back to contributing.

**Comment by brettcannon** [comment]
@merwok Invite sent! (and sorry to @feanil for the noise on his PR 😄 )

File: Lib/test/test_pydoc.py
Comment (feanil) [reply Lib/test/test_pydoc.py]: @brettcannon just making sure you don't want me to do anything here? The one place that 'localhost' is used in pydoc.py is for the default vaule of the hostname parameter for the `browse()` function to not break backwards compatibility.

**Comment by feanil** [comment]
I didn't expect the Spanish Inquisition!

File: Lib/test/test_pydoc.py
Comment (merwok) [reply Lib/test/test_pydoc.py]: See my comment before Brett’s: technically, this test does not prove that the patch is correct, since `'localhost'` is the value that was used before your changes.

File: Misc/ACKS
Comment (merwok) [inline Misc/ACKS]: This file is (mostly) sorted, can you move your name up with the other P?

**Review by merwok — CHANGES_REQUESTED** [review]

**Comment by merwok** [comment]
By the way, is the command-line interface documented in the reST docs?  If so, please add the new option there too.  You could mention that it’s useful in containers, since that was the motivating use case.

**Comment by feanil** [comment]
I didn't expect the Spanish Inquisition!

**Review by merwok — APPROVED** [review]

**Pull request merged**
